src/PyListProxyHandler.cc

```diff
@@ -29,7 +30,28 @@
 
 const char PyListProxyHandler::family = 0;
 
+// private util
+// if function is a proxy for a python method, mutate it into a new python method bound to thisObject
+static bool makeNewPyMethod(JSContext *cx, JS::MutableHandleValue function, JS::HandleObject thisObject) {
+  if (!JS_IsNativeFunction(&(function.toObject()), callPyFunc)) {
+    return true; // we don't need to mutate function if it is not a proxy for a python function
+  }
+
+  PyObject *method = (PyObject *)js::GetFunctionNativeReserved(&(function.toObject()), 0).toPrivate();
+  if (!PyMethod_Check(method)) {
+    PyErr_Format(PyExc_TypeError, "unbound python functions do not have a 'self' to bind");
+    return false;
+  }
 
+  PyObject *func = PyMethod_Function(method);
```

Comment:
are we properly managing the PyObject references around here?

Reply:
yes, PyMethod_Function gives a borrowed reference to func, and PyMethod_New gives a new reference to a python method (which will eventually be decrefed by functionRegistryCallback.